core/views.py

```
@@ -635,3 +656,380 @@ def post(self, request, *args, **kwargs):
+        else:
+            template_name = ""
+
+        return template_name
+
+    def get_initial(self):
+        initial = super().get_initial()
+        if self.request.GET.get("conflict"):
+            initial["incoming_contact"] = self.request.GET.get("conflict")
+
+        if (
+            self.request.GET.get("method")
+            == ConflictResolutionMethods.KEEP_OLD_DATA.value
+        ):
+            initial["method"] = ConflictResolutionMethods.KEEP_OLD_DATA.value
+
+        if (
+            self.request.GET.get("method")
+            == ConflictResolutionMethods.SAVE_INCOMING_DATA.value
+        ):
+            initial["method"] = ConflictResolutionMethods.SAVE_INCOMING_DATA.value
+
+        return initial
+
+    def get(self, request, *args, **kwargs):
+        if self.request.htmx:
+            return super().get(request, *args, **kwargs)
+        else:
+            raise Http404
+
+    def form_valid(self, form):
+        if form.cleaned_data.get("method") == ConflictResolutionMethods.KEEP_OLD_DATA:
+            incoming_contact_id = form.cleaned_data.get("incoming_contact")
+            TemporaryContact.objects.filter(id=incoming_contact_id).delete()
+        elif (
+            form.cleaned_data.get("method")
+            == ConflictResolutionMethods.SAVE_INCOMING_DATA
+        ):
+            incoming_contact_id = form.cleaned_data.get("incoming_contact")
+            incoming_contact = (
+                TemporaryContact.objects.filter(id=incoming_contact_id)
+                .select_related("record")
+                .first()
+            )
+            try:
+                record = incoming_contact.record
+                update_values = vars(incoming_contact)
+                incoming_contact.delete()
+                update_values.pop("record_id")
+                update_values.pop("id")
+                update_object(record, update_values)
+
+            except Exception as e:
+                print(e)
```

Leftover print and catch all Exception here need to be removed.